src/components/auth/EthWalletNewSelector.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { ProviderType } from '@dcl/schemas/dist/dapps/provider-type'
+import { isCucumberProvider, isDapperProvider } from 'decentraland-dapps/dist/lib/eth'
+import { Loader } from 'decentraland-ui/dist/components/Loader/Loader'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+// import { LoginModal, LoginModalOptionType } from '../LoginModal/LoginModal'
+import { LoginModal, LoginModalOptionType } from 'decentraland-ui/dist/components/LoginModal/LoginModal'
+import { track } from '../../utils/tracking'
+import { EthConnectAdvice } from './EthConnectAdvice'
+
+export interface WalletSelectorProps {
+  open: boolean
+  loading: boolean
+  canceling?: boolean
+  provider?: ProviderType
+  availableProviders?: ProviderType[]
+  onLogin?: (provider: ProviderType) => void
+  onCancelLogin?: () => void
+  onClose?: () => void
+}
+
+export const EthWalletNewSelector: React.FC<WalletSelectorProps> = React.memo(
+  ({ open, loading, canceling, provider, availableProviders, onLogin, onCancelLogin, onClose }) => {
+    const anchor = useMemo(() => {
+      const a = document.createElement('a')
+      a.target = '_blank'
+      a.rel = 'noreferrer noopener'
+      a.href = 'https://metamask.io/download.html'
+      return a
+    }, [])
+
+    const [showIncompatible, setShowIncompatible] = useState(false)
+
+    const browserWallet = useMemo<LoginModalOptionType | null>(() => {
+      if (availableProviders && availableProviders.length > 0) {
+        const hasWallet = availableProviders.includes(ProviderType.INJECTED)
+        if (hasWallet) {
+          return isCucumberProvider()
+            ? LoginModalOptionType.SAMSUNG
+            : isDapperProvider()
+            ? LoginModalOptionType.DAPPER
+            : LoginModalOptionType.METAMASK
+        }
+      }
+
+      return null
+    }, [availableProviders])
+
+    const handleLoginWalletConnect = useCallback(() => onLogin && onLogin(ProviderType.WALLET_CONNECT), [onLogin])
+    const handleLoginWalletLink = useCallback(() => onLogin && onLogin(ProviderType.WALLET_LINK), [onLogin])
+    const handleLoginFortmatic = useCallback(() => onLogin && onLogin(ProviderType.FORTMATIC), [onLogin])
```

Comment:
Can `onLogin` be undefined? If it can, we shouldn't render a button to log in, if it can't we should remove the condition to check if it is.